langchain/src/chat_models/baiduwenxin.ts

```diff
@@ -0,0 +1,459 @@
+   * multiple choice, and temp closer to 1 for creative
+   * and generative tasks. Defaults to 0.95.
+   * Only supported for `modelName` of `WenxinModelName.ERNIE_BOT`.
+   */
+  temperature?: number;
+
+  /** Total probability mass of tokens to consider at each step. Range
+   * from 0 to 1.0. Defaults to 0.8.
+   * Only supported for `modelName` of `WenxinModelName.ERNIE_BOT`.
+   */
+  topP?: number;
+
+  /** Penalizes repeated tokens according to frequency. Range
+   * from 1.0 to 2.0. Defaults to 1.0.
+   * Only supported for `modelName` of `WenxinModelName.ERNIE_BOT`.
+   */
+  penaltyScore?: number;
+}
+
+function messageTypeToWenxinRole(type: MessageType): WenxinMessageRole {
+  switch (type) {
+    case "ai":
+      return "assistant";
+    case "human":
+      return "user";
+    case "system":
+      throw new Error("System messages not supported");
+    default:
+      throw new Error(`Unknown message type: ${type}`);
+  }
+}
+
+/**
+ * Wrapper around Baidu ERNIE large language models that use the Chat endpoint.
+ *
+ * To use you should have the `BAIDU_API_KEY` and `BAIDU_SECRET_KEY`
+ * environment variable set.
+ *
+ * @augments BaseLLM
+ * @augments BaiduERNIEInput
+ */
+export class ChatBaiduWenxin
+  extends BaseChatModel
+  implements BaiduWenxinChatInput
+{
+  declare CallOptions: BaseLanguageModelCallOptions;
+
+  get callKeys(): string[] {
+    return ["stop", "signal", "options"];
+  }
+
+  get lc_secrets(): { [key: string]: string } | undefined {
+    return {
+      baiduApiKey: "BAIDU_API_KEY",
+      baiduSecretKey: "BAIDU_SECRET_KEY",
+    };
+  }
+
+  get lc_aliases(): { [key: string]: string } | undefined {
+    return undefined;
+  }
+
+  lc_serializable = true;
+
+  baiduApiKey?: string;
+
+  baiduSecretKey?: string;
+
+  accessToken: string;
+
+  streaming?: boolean;
+
+  prefixMessages?: WenxinMessage[];
+
+  userId?: string;
+
+  modelName = "ERNIE-Bot-turbo";
+
+  apiUrl: string;
+
+  temperature?: number | undefined;
+
+  topP?: number | undefined;
+
+  penaltyScore?: number | undefined;
+
+  constructor(fields?: Partial<BaiduWenxinChatInput> & BaseChatModelParams) {
+    super(fields ?? {});
+
+    this.baiduApiKey =
+      fields?.baiduApiKey ?? getEnvironmentVariable("BAIDU_API_KEY");
+    if (!this.baiduApiKey) {
+      throw new Error("Baidu API key not found");
+    }
+
+    this.baiduSecretKey =
+      fields?.baiduSecretKey ?? getEnvironmentVariable("BAIDU_SECRET_KEY");
+    if (!this.baiduSecretKey) {
+      throw new Error("Baidu Secret key not found");
+    }
+
+    this.streaming = fields?.streaming ?? this.streaming;
+    this.prefixMessages = fields?.prefixMessages ?? this.prefixMessages;
+    this.userId = fields?.userId ?? this.userId;
+
+    this.modelName = fields?.modelName ?? this.modelName;
+
+    switch (this.modelName) {
+      case "ERNIE-Bot":
+        this.apiUrl =
+          "https://aip.baidubce.com/rpc/2.0/ai_custom/v1/wenxinworkshop/chat/completions";
```

Comment:
Is `aip` a typo? 

Can you point me towards the docs for Ernie? I can't seem to find any on Google...

Reply:
It is not a typo. When the first time I read this at official site, I have exact the same question with you.

Docs:
1. ERNIE Bot: https://cloud.baidu.com/doc/WENXINWORKSHOP/s/jlil56u11
2. ERNIE Bot turbo: https://cloud.baidu.com/doc/WENXINWORKSHOP/s/4lilb2lpf

Reply:
Thank you!